src/inference/dev_api/openvino/runtime/iplugin.hpp

```diff
@@ -0,0 +1,239 @@
+ */
+class OPENVINO_RUNTIME_API IPlugin : public std::enable_shared_from_this<IPlugin> {
+public:
+    /**
+     * @brief Sets a plugin version
+     *
+     * @param version A version to set
+     */
+    void set_version(const Version& version);
+
+    /**
+     * @brief Returns a plugin version
+     *
+     * @return A constant ov::Version object
+     */
+    const Version& get_version() const;
+
+    /**
+     * @brief Sets a name for the plugin
+     *
+     * @param name Plugin name
+     */
+    void set_device_name(const std::string& name);
+
+    /**
+     * @brief Provides a plugin name
+     *
+     * @return Plugin name
+     */
+    const std::string& get_device_name() const;
+
+    /**
+     * @brief Compiles model from ov::Model object
+     * @param model A model object acquired from ov::Core::read_model or source construction
+     * @param properties A ov::AnyMap of properties relevant only for this load operation
+     * @return Created Compiled Model object
+     */
+    virtual std::shared_ptr<InferenceEngine::IExecutableNetworkInternal> compile_model(
+        const std::shared_ptr<const ov::Model>& model,
+        const ov::AnyMap& properties) const = 0;
+
+    /**
+     * @brief Compiles model from ov::Model object
+     * @param model_path A path to model (path can be converted from unicode representation)
+     * @param properties A ov::AnyMap of properties relevant only for this load operation
+     * @return Created Compiled Model object
+     */
+    virtual std::shared_ptr<InferenceEngine::IExecutableNetworkInternal> compile_model(
+        const std::string& model_path,
+        const ov::AnyMap& properties) const = 0;
+
+    /**
+     * @brief Compiles model from ov::Model object, on specified remote context
+     * @param model A model object acquired from ov::Core::read_model or source construction
+     * @param properties A ov::AnyMap of properties relevant only for this load operation
+     * @param context A pointer to plugin context derived from RemoteContext class used to
+     *        execute the model
+     * @return Created Compiled Model object
+     */
+    virtual std::shared_ptr<InferenceEngine::IExecutableNetworkInternal> compile_model(
+        const std::shared_ptr<const ov::Model>& model,
+        const ov::AnyMap& properties,
+        const ov::RemoteContext& context) const = 0;
+
+    /**
+     * @brief Sets properties for plugin, acceptable keys can be found in openvino/runtime/properties.hpp
+     * @param properties ov::AnyMap of properties
+     */
+    virtual void set_property(const ov::AnyMap& properties);
+
+    /**
+     * @brief Gets properties related to plugin behaviour.
+     *
+     * @param name Property name.
+     * @param arguments Additional arguments to get a property.
+     *
+     * @return Value of a property corresponding to the property name.
+     */
+    virtual ov::Any get_property(const std::string& name, const ov::AnyMap& arguments) const;
```

Comment:
why set / get properties have default implementation which throws? As far as I remember, we agreed that plugin developers needs to throw explicitly

Reply:
I would like to add default implementation in the next PR.